### PR TITLE
Use test_random() in hashtable mt test for better reproducibility

### PR DESCRIPTION
**Pull request by t8m**
Although due to to the multithreading, it won't be perfect, it might help.

### CODE REVIEW

**Review by Hamidghazimorad — APPROVED** [review]

**Comment by nhorman** [comment]
what issue is this addressing?

**Comment by t8m** [comment]
> what issue is this addressing?

I am trying to reproduce this failure:
https://github.com/openssl/openssl/actions/runs/11292741961/job/31409362706

File: test/lhash_test.c [hunk 1]
```diff
@@ -555,16 +557,11 @@ static void do_mt_hash_work(void)
...
             return;
-        }
-        index %= TEST_MT_POOL_SZ;
-        if (!RAND_bytes((unsigned char *)&behavior, sizeof(char))) {
-            worker_exits[num] = "Failed to get random behavior";
-            return;
-        }
-        behavior &= BEHAVIOR_MASK;
+        index = test_random() % TEST_MT_POOL_SZ;
+        behavior = (char)(test_random() & BEHAVIOR_MASK);
```
Comment (paulidale) [inline test/lhash_test.c:563]: Suggest using `%` here to avoid confusing breaks when the macro is changed.

**Review by paulidale — APPROVED** [review]

File: test/lhash_test.c (hunk 1 above)
Comment (t8m) [reply test/lhash_test.c]: Done

Comment (t8m) [reply test/lhash_test.c]: Please re-approve

File: test/lhash_test.c [hunk 2]
```diff
@@ -528,16 +530,16 @@ static void hashtable_mt_free(HT_VALUE *v)
     }
 }
 
-#define BEHAVIOR_MASK 0x3
 #define DO_LOOKUP 0
 #define DO_INSERT 1
 #define DO_REPLACE 2
 #define DO_DELETE 3
+#define NUM_BEHAVIORS (DO_DELETE + 1)
```
Comment (paulidale) [inline test/lhash_test.c:537]: An enum would seem better suited here, but it's not critical.

**Review by paulidale — APPROVED** [review]

**Review by hlandau — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master and 3.4 branches. Thank you for the reviews.

**Pull request closed without merging**